kyo-pure/shared/src/main/scala/kyo/streams.scala

```diff
@@ -0,0 +1,265 @@
+package kyo
+
+import kyo.*
+import kyo.core.*
+import kyo.core.internal.*
+import scala.annotation.tailrec
+
+opaque type Stream[+T, V, -S] = T < (Streams[V] & S)
```

Comment:
Removed some methods that depend on `IOs` to handle `Channels`. The idea is to reintroduce them as extension methods in `kyo-core`.